ts/main.ts

```diff
@@ -39,24 +43,45 @@ function handleInput(): void {
 
 function handleSubmit(event: SubmitEvent): void {
   event.preventDefault();
-  const newFormEntry: CodeJournalForm = {
-    entryId: data.nextEntryId,
-    title: $entryTitle.value,
-    photoURL: $photoURL.value,
-    notes: $entryNotes.value,
-  };
-  data.nextEntryId += 1;
-  data.entries.unshift(newFormEntry);
+  if (data.editing !== null) {
+    const entryIndex = data.entries.findIndex(
+      (entry) => entry.entryId === data.editing!.entryId,
+    );
+    data.entries[entryIndex] = {
+      entryId: data.editing.entryId,
+      title: $entryTitle.value,
+      photoURL: $photoURL.value,
+      notes: $entryNotes.value,
+    };
+    $form.reset();
+    $img!.src = 'images/placeholder-image-square.jpg';
+    $entryNotes.textContent = '';
```

Comment:
`$form.reset()` should already be resetting the form, so this shouldn't be necessary